src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/AddUsingsCodeActionResolverTest.cs

```diff
@@ -0,0 +1,470 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.CodeActions;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Text;
+using Moq;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using Microsoft.AspNetCore.Razor.Language.Extensions;
+using Microsoft.AspNetCore.Mvc.Razor.Extensions;
+using System.Runtime.ExceptionServices;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.CodeActions
+{
+    public class AddUsingsCodeActionResolverTest : LanguageServerTestBase
+    {
+        private readonly DocumentResolver EmptyDocumentResolver = Mock.Of<DocumentResolver>();
+
+        [Fact]
+        public async Task Handle_MissingFile()
+        {
+            // Arrange
+            var resolver = new AddUsingsCodeActionResolver(new DefaultForegroundDispatcher(), EmptyDocumentResolver);
+            var data = JObject.FromObject(new CreateComponentCodeActionParams()
+            {
+                Uri = new Uri("c:/Test.razor"),
+                Path = "c:/Another.razor",
+            });
+
+            // Act
+            var workspaceEdit = await resolver.ResolveAsync(data, default);
+
+            // Assert
+            Assert.Null(workspaceEdit);
+        }
+
+        [Fact]
+        public async Task Handle_Unsupported()
+        {
+            // Arrange
+            var documentPath = "c:/Test.razor";
+            var contents = $"@page \"/test\"";
+            var codeDocument = CreateCodeDocument(contents);
+            codeDocument.SetUnsupported();
+
+            var resolver = new AddUsingsCodeActionResolver(new DefaultForegroundDispatcher(), CreateDocumentResolver(documentPath, codeDocument));
+            var data = JObject.FromObject(new CreateComponentCodeActionParams()
+            {
+                Uri = new Uri(documentPath),
+                Path = "c:/Another.razor",
+            });
+
+            // Act
+            var workspaceEdit = await resolver.ResolveAsync(data, default);
+
+            // Assert
+            Assert.Null(workspaceEdit);
+        }
+
+        [Fact]
+        public async Task Handle_InvalidFileKind()
+        {
+            // Arrange
+            var documentPath = "c:/Test.razor";
+            var contents = $"@page \"/test\"";
```

Comment:
`$` is for string interpolation when you utilize `{variables}` . Don't need $ in front of all your strings that don't do that in this class

Reply:
Yeah this is just a copy paste typo :)